OpenWeatherApp/ViewControllers/SoftAskViewController.swift

```diff
@@ -0,0 +1,162 @@
+      ])
+  }
+  private func setupTopView() {
+    containerView.addSubview(topContainerView)
+    NSLayoutConstraint.activate([
+      topContainerView.topAnchor.constraint(equalTo: containerView.topAnchor),
+      topContainerView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
+      topContainerView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
+      topContainerView.heightAnchor.constraint(equalTo: view.heightAnchor, multiplier: 0.5)
+      ])
+  }
+  private func setupTopStackView() {
+    let topStackView = UIStackView(arrangedSubviews: [greetingLabel, askLabel])
+    topStackView.axis = .vertical
+    topStackView.translatesAutoresizingMaskIntoConstraints = false
+    topContainerView.addSubview(topStackView)
+    NSLayoutConstraint.activate([
+      topStackView.centerYAnchor.constraint(equalTo: topContainerView.centerYAnchor),
+      topStackView.leadingAnchor.constraint(equalTo: topContainerView.leadingAnchor, constant: 110),
+      topStackView.trailingAnchor.constraint(equalTo: topContainerView.trailingAnchor, constant: -110),
+      topStackView.heightAnchor.constraint(equalTo: topContainerView.heightAnchor, multiplier: 0.3)
+      ])
+  }
+  private func setupBottomStackView() {
+    let bottomStackView = UIStackView(arrangedSubviews: [yesButton, maybeButton])
+    bottomStackView.distribution = .fillEqually
+    bottomStackView.axis = .vertical
+    bottomStackView.translatesAutoresizingMaskIntoConstraints = false
+    containerView.addSubview(bottomStackView)
+    NSLayoutConstraint.activate([
+      bottomStackView.heightAnchor.constraint(equalToConstant: 100),
+      bottomStackView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
+      bottomStackView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
+      bottomStackView.topAnchor.constraint(equalTo: topContainerView.bottomAnchor)
+      ])
+  }
+  // MARK: buttons handle actions
+  @objc private func handleYes() {
+    checkLocationServices()
+    // TODO: go to CityListViewController
+  }
+  @objc private func handleMaybe() {
+    // TODO: go to CityListViewController
+  }
+  // MARK: location functions
+  private func checkLocationServices() {
+    if CLLocationManager.locationServicesEnabled() {
+      setupLocationManager()
+      checkLocationAuthorization()
+    } else {
+      let alert = UIAlertController(title: "Location Services disabled",
+                                    message: "Please enable Location Services in Settings",
+                                    preferredStyle: .alert)
+      let okAction = UIAlertAction(title: "OK", style: .default, handler: nil)
+      alert.addAction(okAction)
+      present(alert, animated: true, completion: nil)
+    }
+  }
+  private func setupLocationManager() {
+    locationManager.delegate = self
+    locationManager.desiredAccuracy = kCLLocationAccuracyBest
+  }
```

Comment:
Eu também adicionaria 1 linha vazia entre cada definição de propriedade/função

Reply:
Mas eu teria que alterar a regra no SwiftLint?

Reply:
Hmm creio que a regra exige uma linha de distancia entre as definições, não?

Reply:
Tem sim... eu prefiro dar um espaço também, mas to usando o default. Quer que eu altere?

Reply:
Engraçado... pq o repositório das regras que o SwiftLint segue por default mostra todos os exemplos com espaço entre as definições. (https://github.com/github/swift-style-guide)

Reply:
Enfim, se puder adicionar essa regra acho que o código fica menos espremido :)

Reply:
O que estava acontecendo é que ele estava acusando que tinha espaço em branco na linha, mas não estava reclamando da linha em si.

Reply:
Só remover o check dessa preferencia:

`Xcode > Preferences`
<img width="803" alt="screen shot 2019-02-22 at 15 50 14" src="https://user-images.githubusercontent.com/5031753/53263955-95b49400-36b9-11e9-8726-c288affceb94.png">